wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/watcher/server/detectors/ProjectTreeTracker.java

```diff
@@ -154,9 +170,41 @@ public void configureHandler(RequestHandlerConfigurator configurator) {
             .paramsAsDto(params)
             .sendAndSkipResult();
       }
+
+      fireCreatedEventIfIsProject(it);
     };
   }
 
+  private void fireCreatedEventIfIsProject(String path) {
+    // Two or more clients can be subscribed to this handler, so need to skip firing the event
+    // if it was already sent.
+    if (registeredProjectPaths.contains(path)) {
+      return;
+    }
+
+    // Check if the given folder is a project.
+    final Timer timer = new Timer();
```

Comment:
I don't understand why we need to do this with a timer. What is the scenario that we are covering with the delay?

Reply:
This detects project path creation, but we need to wait some time for the project to be registered. If the number of attempts will be exceeded it will mean that the given path is just a new folder.

Reply:
But then the ProjectCreatedEvent should be sent by whoever creates the project.